src/index.ts

```diff
@@ -207,13 +211,33 @@ function startApp(appSchema, path: string) {
     }
   })
 
+  const graphqlUpload = graphqlUploadExpress({
+    maxFileSize: GRAPHQL_UPLOAD_MAX_FILE_SIZE_IN_BYTES,
+    maxFiles: GRAPHQL_UPLOAD_MAX_FILES,
+  })
+
   app.use("/batch", bodyParser.json(), graphqlBatchHTTPWrapper(graphqlServer))
-  app.use(graphqlServer)
+  app.use(graphqlUpload, graphqlServer)
 
   if (enableSentry) {
     app.use(Sentry.Handlers.errorHandler())
   }
 
+  // Global error handler
+  app.use((error, _req, res, next) => {
+    if (res.headersSent) {
+      return next(error)
+    }
+
+    const status = error?.status ?? 500
+    const message = error?.message ?? "Something went wrong"
+
+    return res.status(status).json({
+      message,
+      stack: !PRODUCTION_ENV ? error.stack.split("\n") : null,
+    })
+  })
+
```

Comment:
**question**: Should global error handler be used in Express to return error message and stack trace (_see screenshot 1, 2_)? Otherwise, we will always get 404 status code and the corresponding message text for this error (_see attached screenshot 3_).

On the one hand, it can help developers to quickly understand what is working wrong and find error. For example, as it was in my case when I started working with `graphql-upload` and always got `404 Cannot POST /v2` in response to requests where I sent the file. It turned out that I used the wrong query structure and an error was thrown in the `graphql-upload` middleware for this reason. But due to the fact that we did not have a global error handler, the default `404 error` was always returned.

But on the other hand, won't I break something in **another place or in another application** if I add these changes?

What do you think about this? Or maybe add a global error handler only for staging/development environments?

#### Screenshot 1
<img width="1630" alt="error-1" src="https://user-images.githubusercontent.com/3513494/164517982-bfd648f3-7a25-4b68-b0ed-01ea75732aeb.png">


#### Screenshot 2
<img width="1630" alt="error-2" src="https://user-images.githubusercontent.com/3513494/164517996-a307f6b3-e21f-44e7-88c8-eb97365d5f83.png">

#### Screenshot 3 (without global error handler)
<img width="1249" alt="error-3" src="https://user-images.githubusercontent.com/3513494/164518113-a06da037-f5f3-431d-97c4-31db3f91711e.png">

Reply:
Interesting that 404 is the default error response. I guess that might be because we don't currently respond to `multipart/form-data` requests?

This sounds like a nice change but I'd suggest splitting it out into its own PR where we can discuss the implementation separate from this spike for image uploading. Some early feedback though: I might extract the handler into a separate file and perhaps have a specific environment variable to explicitly enable the middleware and verbose error responses.

Reply:
> This sounds like a nice change but I'd suggest splitting it out into its own PR where we can discuss the implementation separate from this spike for image uploading

I agree. I'll do it